run_project_tests.py

```diff
@@ -675,6 +675,11 @@ def detect_system_compiler():
     else:
         raise RuntimeError("Could not find C compiler.")
 
+def setup_pythonpath():
+    # Make sure python can import mesonbuild.
+    test_dir = os.path.dirname(os.path.abspath(__file__))
+    os.environ['PYTHONPATH'] = test_dir
+
```

Comment:
This is not correct due to two reasons:

- the tests are only guaranteed to work when run from the source root so when using Meson in the current source dir, the path is already set up
- in e.g. Debian packaging tests the Meson executable to use comes from the system and it _must not_ use any Python modules from the source root

Reply:
**Unit tests**: I agree that we shouldn't need `PYTHONPATH` since `.` is implicitly in it. Most of the tests run just fine without it. However, we run a bunch of commands outside of the source directory during some of the tests (e.g. `test_dist_git`). Those commands fail because they can't find the modules in the source directory (for example, if ninja executes `python3 -m mesonbuild.mesonmain` from a tmp directory). We can either add the source directory to `PYTHONPATH` during the tests so that everything "just works", or carefully pass `-C` to every command we run outside of the source directory. I'm OK with either one of these; let me know what you prefer.

**Project tests**: If meson has already been installed, then this should work just fine. However, I added this code to fix breakages I saw in CI. When I install on my own machine using setup.py, I'm not seeing those breakages on my own machine. So, I'll remove this code and we can try to figure out why the CI machines are stumbling over it.

Reply:
OK, I pushed a change to remove the `PYTHONPATH` adjustment from the project tests; I expect the CI will break, and we need to debug that.